tests/components/arcam_fmj/test_media_player.py

```diff
@@ -0,0 +1,348 @@
+"""Tests for arcam fmj receivers."""
+from math import isclose
+
+from arcam.fmj import DecodeMode2CH, DecodeModeMCH, IncomingAudioFormat, SourceCodes
+from asynctest.mock import ANY, MagicMock, Mock, PropertyMock, patch
+import pytest
+
+from homeassistant.components.arcam_fmj.const import (
+    DOMAIN,
+    SIGNAL_CLIENT_DATA,
+    SIGNAL_CLIENT_STARTED,
+    SIGNAL_CLIENT_STOPPED,
+)
+from homeassistant.components.arcam_fmj.media_player import ArcamFmj
+from homeassistant.components.media_player.const import MEDIA_TYPE_MUSIC
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+
+from .conftest import MOCK_HOST, MOCK_NAME, MOCK_PORT
+
+MOCK_TURN_ON = {
+    "service": "switch.turn_on",
+    "data": {"entity_id": "switch.test"},
+}
+
+
+async def test_properties(player, state):
+    """Test standard properties."""
+    assert player.unique_id is None
```

Comment:
We shouldn't assert anything about the entity directly. We should always use a core representation of the entity state or its attributes. Normally this is the state of the entity in the state machine, in some cases we need to check the entity registry entry of the entity.

Writing the tests like this will make them more robust for changes in the platform.

Reply:
I'll see if i can rewrite it. I somewhat disagree on the change thou. Now it's unit testing, if run it through core, it's integration testing.

Reply:
Yes. The unit testing isn't helpful since we only care about state consistency. The entity implementation should be able to change as long as the state remains the same.